src/engine/QueryExecutionTree.cpp

```diff
@@ -210,7 +215,12 @@ nlohmann::json QueryExecutionTree::writeJsonTable(
   for (size_t i = from; i < upperBound; ++i) {
     json.emplace_back();
     auto& row = json.back();
-    for (const auto& idx : validIndices) {
+    for (const auto& opt : validIndices) {
+      if (!opt) {
+        row.emplace_back(optToJson(std::optional<string>{}));
```

Comment:
Couldn't you just feed in `nullptr` here given that the optional is never set? That i what the optToJson method does as well after all. If you don't want to define json null twice you could define a null constant together with the lambda and use that in both.
If you definitely want to use the lambda you could use std::nullopt to make it clear that the optional does not hold a value.

Reply:
Yes, I would leave the format as is for now.
This is only intended to be a bugfix. The Json format is relatively verbose after all and probably not intended for
large results, so this is not super urgend imho.